saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/upload/EnrichmentTagUploadService.java

```diff
@@ -103,15 +123,27 @@ public void uploadEnrichmentTags(InputStream inputStream) {
             throw new ServiceException(ERROR_IO, e.getMessage(), e);
         }
          
-        catch (DataIntegrityViolationException e) {
-            throw new ServiceException(ERROR_SAME_TAG_ALREADY_UPLOADED, e.getMessage(), e);
+        catch (PersistenceException e) {
+        	if (e.getCause() instanceof ConstraintViolationException) {
+        		throw new ServiceException(ERROR_SAME_TAG_ALREADY_UPLOADED, e.getCause().getMessage(), e.getCause());
+        	} else {
+        		throw new ServiceException(ERROR_INTERVAL_SERVER_ERROR, e.getMessage(), e.getCause());
+        	}
         }
         
     }
 
    
         
-    
+    //------------------------ GETTERS --------------------------
+    public long getEnrichmentTagMaxUploadSize() {
+		return enrichmentTagMaxUploadSize;
+	}
+
```

Comment:
In wiki section GETTERS is before LOGIC.
But I think the way you did it is nicer.
Maybe we should discuss about order of sections?